Change bootloader-id to work correctly after grub boothole v2 updates

mmlb opened the pull request:
## Description

Makes osie install uefi with --bootloader-id=ubuntu.

## Why is this needed

Recent Ubuntu grub updates to fix the BootHole vulnerabilities ended up
hard coding the grub config lookup from \efi\ubuntu, but `grub-install --bootloader-id=grub`
places it into `\efi\grub`. Boot failures ensue.

## How Has This Been Tested?

Modified `ci/vm.sh` to boot the installed VM.
I was able to reproduce the boot hang without this change and no issues after.

mmlb commented:
Failing for centos_7 atm

tstromberg commented on docker/scripts/grub-installer.sh:
```diff
@@ -141,7 +141,7 @@ install_grub_osie() {
 		else
 			# grub-install doesn't error if efibootmgr can't actually set the boot entries/order
-			efibootmgr | tee /dev/stderr | grep -iq grub
+			efibootmgr | tee /dev/stderr | grep -iq ubuntu
```
Neat! I've never seen the `tee /dev/stderr` trick. The comment I had was this can be fragile, and break in certain obscure environments (chroots and what not). The native way to do this is in shells (at least those produced after 1993) is:

`efibootmgr 2>&1 | grep -iq ubuntu`

Feel free to ignore this suggestion in preference to putting it into a refactor-only PR.

mmlb replied:
The `tee` is here so we can see all of `efibootmgr`'s output on stderr while copying to stdout and into `grep`. `efitbootmgr 2>&1` won't duplicate to stderr. This is really just a hack to get around running `efibootmgr` twice, which doesn't really buy us anything really, I'm fine with calling it twice.

tstromberg commented on docker/scripts/grub-installer.sh:
```diff
@@ -126,7 +126,7 @@ install_grub_osie() {
 		[[ $arch == aarch64 ]] && mount -o remount,ro /sys/firmware/efi/efivars
 
-		grub-install --recheck --bootloader-id=GRUB --root-directory="$target" --efi-directory="$target/boot/efi"
+		grub-install --recheck --bootloader-id=ubuntu --root-directory="$target" --efi-directory="$target/boot/efi"
```
Since it seems like the bootloader_id has to exactly match between here and the later `efibootmgr` line later - could you please extract it into a variable?

`readonly BOOTLOADER_ID=ubuntu`

mmlb replied:
Not sure if this is going to stay this way to get centos (and presumably debian and others) working. I'll get rid of the duplication if it happens to stick around though.

mmlb committed:
ci/vm: Sort and align json in gen_metadata

Signed-off-by: Manuel Mendez <mmendez@equinix.com>

mmlb committed:
ci/vm: Fix/tweak Caddyfile

Signed-off-by: Manuel Mendez <mmendez@equinix.com>

mmlb committed:
ci/vm: Log caddy http requests to a file

Is helpful to debug the tests.

Signed-off-by: Manuel Mendez <mmendez@equinix.com>

mmlb committed:
ci/vm: Always use virtio-net nic

No need to emulate e1000 when we have virtio-net drivers
available.

Signed-off-by: Manuel Mendez <mmendez@equinix.com>

mmlb committed:
ci/vm: Pull out qemu diret boot args from run_vm

This will make run_vm reusable for booting the provisioned VM.

Signed-off-by: Manuel Mendez <mmendez@equinix.com>

mmlb committed:
ci/vm: Get rid of `plan` from metadata

Plan has been deprecated for years, lets finally drop it.

Signed-off-by: Manuel Mendez <mmendez@equinix.com>

mmlb committed:
ci/vm: Drop do_network_test

We no longer carry the packet-networking scripts that this was testing

Signed-off-by: Manuel Mendez <mmendez@equinix.com>

mmlb committed:
ci/vm: Verify cmd passed in through cmdline is known.

Signed-off-by: Manuel Mendez <mmendez@equinix.com>

mmlb committed:
ci/vm: Change shebang to /usr/bin/env bash

Helps for when trying to run under nixos.

Signed-off-by: Manuel Mendez <mmendez@equinix.com>

mmlb committed:
ci/vm: Move metadata disk renaming into gen_metadata

No need to have that complicated sed stuff popping out in do_test, its
internal info anyway.

Signed-off-by: Manuel Mendez <mmendez@equinix.com>

mmlb committed:
ci/vm: Avoid different types for bios variable

No real good reason for mixing array vs string, bash won't treat the empty
array as unset when in non-UEFI mode.

Signed-off-by: Manuel Mendez <mmendez@equinix.com>

mmlb committed:
ci/vm: Setup qemu kvm with 2 serial ports, console on the second one

Just like most bare metal machines. This will also come in handy soon
when testing booting the installed vm and having serial console setup
correctly.

Signed-off-by: Manuel Mendez <mmendez@equinix.com>

mmlb committed:
ci/vm: Refactor actual test in do_test into test_provision

do_test will basically just call each of the specific tests.

Signed-off-by: Manuel Mendez <mmendez@equinix.com>

mmlb committed:
ci/vm: Add test_deprovision

This is similar to test_provision but checks for the deprovisioning events
messages.

Signed-off-by: Manuel Mendez <mmendez@equinix.com>

mmlb committed:
ci/vm: Only randomize nic order/link status once

Now that we run the VM multiple times we shouldn't be randomizing
the nic macs and link status per run, this isn't really very realistic.

Signed-off-by: Manuel Mendez <mmendez@equinix.com>

mmlb committed:
ci/vm: Add some color when running the vm tests

Now that we're running more than one VM boot based test having the test
name in the prefix helps with knowing which test is being run. The color
helps even more, especially by ensuring a color reset happens on each line.
Piping to another process also combats cursor movement/reset codes that
happen on boot and clear history which makes debugging a pain.

Signed-off-by: Manuel Mendez <mmendez@equinix.com>

mmlb committed:
wip

tstromberg commented:
Is this PR ready to be re-reviewed?

mmlb commented:
Nope, closing in favor of #228
